requirements.txt

```diff
@@ -1,7 +1,7 @@
 setuptools
 numpy
 scipy
-scikit-learn<=0.19.2
+scikit-learn
```

Comment:
Maybe we should set the version to the earliest version that our CI can pass. It appears to me it should be something like `scikit-learn >= 0.23.0`? Will 0.22.0 work?

Thanks.

Reply:
@yangarbiter

We only set **`scikit-learn >= 0.24`**.

There is a **conflict** between `0.22---0.23` and `0.24` due to `cluster.KMeans` change.
- Change [random_state in `0.22`](https://github.com/scikit-learn/scikit-learn/pull/9288).
- Change [random_state back in `0.24`](https://scikit-learn.org/dev/whats_new/v0.24.html#changed-models)
But I think a change in `0.24` is a little strange without a detailed discussion on the scikit-learn release.

PS: From `0.23` to `0.24` is a big change. Most APIs are refactored. I wonder maybe they miss something during upgrading.